roles/couchbase-infra/tasks/main.yml

```diff
@@ -0,0 +1,209 @@
+  when: couchbase.servers.size > 0
+
+- name: Create Cluster Role for the user
+  k8s:
+    definition: "{{ lookup('file', 'cluster-role-user.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the User
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRoleBinding
+      metadata:
+        name: couchbasecluster-rolebinding
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: ClusterRole
+        name: couchbasecluster
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRole for the Benchmark User
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: benchmark-operator-for-cb
+        namespace: '{{ meta.namespace }}'
+      rules:
+      - apiGroups:
+        - ""
+        resources:
+        - namespaces
+        verbs:
+        - get
+      - apiGroups:
+        - couchbase.com
+        resources:
+        - couchbaseclusters
+        verbs:
+        - '*'
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the Benchmark User
+  k8s:
+    definition:
+      kind: ClusterRoleBinding
+      apiVersion: rbac.authorization.k8s.io/v1
+      metadata:
+        name: benchmark-operator-for-cb
+      subjects:
+      - kind: ServiceAccount
+        name: benchmark-operator
+        namespace: '{{ meta.namespace }}'
+      roleRef:
+        kind: ClusterRole
+        name: benchmark-operator-for-cb
+        apiGroup: rbac.authorization.k8s.io
+  when: couchbase.servers.size > 0
+
+- name: Download Couchbase Operator from operatorhub.io
+  get_url:
+    url: https://www.operatorhub.io/install/couchbase-operator.v1.1.0.yaml
+    dest: /tmp/cboperator.yaml
+  when: couchbase.servers.size > 0
+
+- name: Deploy Couchbase Operator from operatorhub.io
+  k8s:
+    src: /tmp/cboperator.yaml
+  when: couchbase.servers.size > 0
+
+- name: Wait for Couchbase Operator to be Running....
+  k8s_facts:
+    kind: Pod
+    api_version: v1
+    namespace: my-couchbase-enterprise
+    label_selectors:
+      - name = couchbase-operator
+  register: cb_op_pod
+  until: "'Running' in (cb_op_pod | json_query('resources[].status.phase'))"
+  retries: 10
+  delay: 10
+  when: couchbase.servers.size > 0
+
+# Launch Couchbase cluster from operator
+- name: Create the cb-example-auth Auth Secret
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: cb-example-auth
+        namespace: my-couchbase-enterprise
+      type: Opaque
+      data:
+        username: QWRtaW5pc3RyYXRvcg==
+        password: cGFzc3dvcmQ=
+  when: couchbase.servers.size > 0
+
+# NOTE: The community docker images don't work with OpenShift
+#       currently, so we need to auth with the RH catalog
+#       and pull the usable image from there
+#FIXME: This is an awful and insecure way to do this
+- name: Set rawdockerconfigjson Var
```

Comment:
ohokicanreadthis :100:

Reply:
Cleaning this up a bit, but there will still be a pending FIXME that should be turned into an issue.